File: frontend/src/types/agent.ts
export interface AgentStepTrace {
  step_number: number;
  agent_name: string;
  action: string;
  status: string;
  tool_called: string | null;
  observation: string;
  duration_ms: number;
}

export interface AgentRunRequest {
  agent_type: 'job_scout' | 'application_assistant' | 'prep_coach';
  job_id?: string;
  parameters?: Record<string, any>;
}

export interface AgentRunResponse {
  run_id: string;
  agent_type: string;
  status: string;
  summary: string;
  outputs: Record<string, any>;
  traces: AgentStepTrace[];
  groundedness_score: number;
  evidence_citations: string[];
  created_at: string;
}

File: backend/app/schemas/agent.py
"""Pydantic schemas for LangGraph Multi-Agent Studio."""

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class AgentStepTrace(BaseModel):
    step_number: int
    agent_name: str
    action: str
    status: str = "success"
    tool_called: str | None = None
    observation: str
    duration_ms: int = 150


class AgentRunRequest(BaseModel):
    agent_type: str = Field(..., description="job_scout | application_assistant | prep_coach")
    job_id: uuid.UUID | None = None
    parameters: dict[str, Any] = Field(default_factory=dict)


class AgentRunResponse(BaseModel):
    run_id: uuid.UUID
    agent_type: str
    status: str = "completed"
    summary: str
    outputs: dict[str, Any] = Field(default_factory=dict)
    traces: list[AgentStepTrace] = Field(default_factory=list)
    groundedness_score: float = 0.95
    evidence_citations: list[str] = Field(default_factory=list)
    created_at: datetime
